subprojects/language-java/src/integTest/groovy/org/gradle/api/tasks/JavaToolchainBuildOperationsIntegrationTest.groovy

```diff
@@ -67,52 +67,49 @@ class JavaToolchainBuildOperationsIntegrationTest extends AbstractIntegrationSpe
             public class Foo {}
-        """.stripIndent()
+        """
 
         file("src/test/java/FooTest.java") << """
             public class FooTest {
                 @org.junit.Test
                 public void test() {}
             }
-        """.stripIndent()
-
-        def taskPath = ":$task"
+        """
 
         when:
         runWithInstallation(jdkMetadata, task)
-        def events = eventsFor(taskPath)
+        def events = eventsFor(task)
         then:
-        executedAndNotSkipped(taskPath)
+        executedAndNotSkipped(task)
         assertToolchainUsages(events, jdkMetadata, tool)
 
         when:
         runWithInstallation(jdkMetadata, task)
-        events = eventsFor(taskPath)
+        events = eventsFor(task)
         then:
-        skipped(taskPath)
+        skipped(task)
         if (emitsWhenUpToDate) {
             assertToolchainUsages(events, jdkMetadata, tool)
         }
 
         where:
-        task          | tool           | configureToolchain              | emitsWhenUpToDate
-        "compileJava" | "JavaCompiler" | "with java plugin"              | true
-        "compileJava" | "JavaCompiler" | "with per task"                 | true
-        "compileJava" | "JavaCompiler" | "with java plugin and per task" | true
-        "compileJava" | "JavaCompiler" | "without"                       | true
-        "test"        | "JavaLauncher" | "with java plugin"              | true
-        "test"        | "JavaLauncher" | "with per task"                 | true
-        "test"        | "JavaLauncher" | "with java plugin and per task" | true
-        "test"        | "JavaLauncher" | "without"                       | true
-        "javadoc"     | "JavadocTool"  | "with java plugin"              | false // TODO: this must be fixed
-        "javadoc"     | "JavadocTool"  | "with per task"                 | false // TODO: this must be fixed
-        "javadoc"     | "JavadocTool"  | "with java plugin and per task" | false // TODO: this must be fixed
-        "javadoc"     | "JavadocTool"  | "without"                       | false // TODO: this must be fixed
+        task           | tool           | configureToolchain              | emitsWhenUpToDate
+        ":compileJava" | "JavaCompiler" | "with java plugin"              | true
+        ":compileJava" | "JavaCompiler" | "with per task"                 | true
+        ":compileJava" | "JavaCompiler" | "with java plugin and per task" | true
+        ":compileJava" | "JavaCompiler" | "without"                       | true
+        ":test"        | "JavaLauncher" | "with java plugin"              | true
+        ":test"        | "JavaLauncher" | "with per task"                 | true
+        ":test"        | "JavaLauncher" | "with java plugin and per task" | true
+        ":test"        | "JavaLauncher" | "without"                       | true
+        ":javadoc"     | "JavadocTool"  | "with java plugin"              | false // TODO: this must be fixed
```

Comment:
This should work after merging https://github.com/gradle/gradle/pull/21363, right?

Reply:
Not sure, because IIRC it does not work due to `Javadoc` task not having a `JavaVersion` input, like `JavaCompile` and `Test` do.